File: src/test/java/com/github/leomillon/uuidgenerator/EditorDocumentUtilsTest.java
package com.github.leomillon.uuidgenerator;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class EditorDocumentUtilsTest {

    private static final String UUID_NO_DASH = "f228078510824c8baeffbb852eb98783";
    private static final String UUID = "f2280785-1082-4c8b-aeff-bb852eb98783";

    @Test
    void should_remove_dashes_from_uuid() {

        // When
        String result = EditorDocumentUtils.toggleUUIDDashes(UUID);

        // Then
        assertEquals(result, UUID_NO_DASH);
    }
    @Test
    void should_insert_dashes_into_uuid() {

        // When
        String result = EditorDocumentUtils.toggleUUIDDashes(UUID_NO_DASH);

        // Then
        assertEquals(result, UUID);
    }

    @Test
    void should_throw_invalid_format_exception_for_invalid_uuid() {
        assertThrows(InvalidFormatException.class, () -> {
            EditorDocumentUtils.toggleUUIDDashes("f2280785 0824c8baeffbb852eb98783");
        });
    }

}

File: src/test/kotlin/com/github/leomillon/uuidgenerator/EditorDocumentUtilsTest.kt
package com.github.leomillon.uuidgenerator

import assertk.assertThat
import assertk.assertions.isEqualTo
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

class EditorDocumentUtilsTest {

    companion object {

        @Suppress("unused")
        @JvmStatic
        fun uuidWithAndWithoutDashes() = listOf(
            Arguments.of(
                "71bc4adb-1b7c-4827-940c-f1c53a3f7766",
                "71bc4adb1b7c4827940cf1c53a3f7766"
            ),
            Arguments.of(
                "71BC4ADB-1B7C-4827-940C-F1C53A3F7766",
                "71BC4ADB1B7C4827940CF1C53A3F7766"
            )
        )
    }

    @ParameterizedTest
    @MethodSource("uuidWithAndWithoutDashes")
    fun `should remove dashes from uuid`(uuidWithDashes: String, uuidWithoutDash: String) {

        // When
        val result = EditorDocumentUtils.toggleUUIDDashes(uuidWithDashes)

        // Then
        assertThat(result).isEqualTo(uuidWithoutDash)
    }

    @ParameterizedTest
    @MethodSource("uuidWithAndWithoutDashes")
    fun `should insert dashes into uuid`(uuidWithDashes: String, uuidWithoutDash: String) {

        // When
        val result = EditorDocumentUtils.toggleUUIDDashes(uuidWithoutDash)

        // Then
        assertThat(result).isEqualTo(uuidWithDashes)
    }

    @Test
    fun `should throw invalid format exception for invalid uuid`() {

        val error = assertThrows<InvalidFormatException> {
            EditorDocumentUtils.toggleUUIDDashes("71bc4adb 1b7c-4827-940c-f1c53a3f7766")
        }
        assertThat(error.message).isEqualTo("71bc4adb 1b7c-4827-940c-f1c53a3f7766 is not a valid uuid format")
    }
}
